Modules/Registration/Common/include/itkLandmarkBasedTransformInitializer.hxx

```diff
@@ -355,71 +361,180 @@ LandmarkBasedTransformInitializer<TTransform, TFixedImage, TMovingImage>::Intern
+  // Otherwise the versor will be an identity versor.
+  if (m_FixedLandmarks.size() >= ImageDimension)
   {
-    fixedCentroid[0] += (*fixedItr)[0];
-    fixedCentroid[1] += (*fixedItr)[1];
-    fixedCentroid[2] += (*fixedItr)[2];
-    ++fixedItr;
+    itk::Matrix<ParametersValueType, ImageDimension, ImageDimension> M;
+
+    PointsContainerConstIterator fixedItr = m_FixedLandmarks.begin();
+    PointsContainerConstIterator movingItr = m_MovingLandmarks.begin();
+
+    VectorType fixedCentered;
+    VectorType movingCentered;
+
+    fixedCentered.Fill(0.0);
+    movingCentered.Fill(0.0);
+
+    itkDebugStatement(int ii = 0);
+
+    // Computations are relative to the Center of Rotation.
+    while (movingItr != m_MovingLandmarks.end())
+    {
+      for (unsigned int i = 0; i < ImageDimension; ++i)
+      {
+        fixedCentered[i] = (*fixedItr)[i] - fixedCentroid[i];
+        movingCentered[i] = (*movingItr)[i] - movingCentroid[i];
+      }
+
+      for (unsigned int i = 0; i < ImageDimension; ++i)
+      {
+        for (unsigned int j = 0; j < ImageDimension; ++j)
+        {
+          // mmm this indices i,j may have to be reverted...
+          M[i][j] += fixedCentered[i] * movingCentered[j];
+        }
+      }
+
+      itkDebugStatement(++ii);
+      itkDebugMacro(<< "f_" << ii << " = " << fixedCentered);
+      itkDebugMacro(<< "m_" << ii << " = " << movingCentered);
+
+      ++movingItr;
+      ++fixedItr;
+    }
+
+    // -- build the 4x4 matrix N --
+
+    itk::Matrix<ParametersValueType, 4, 4> N;
+
+    CreateMatrix(N, M);
+
+    itkDebugMacro(<< "For Closed form solution: ");
+    itkDebugMacro(<< "M matrix " << M);
+    itkDebugMacro(<< "N matrix " << N);
+
+    vnl_matrix<ParametersValueType> eigenVectors(4, 4);
+    vnl_vector<ParametersValueType> eigenValues(4);
+
+    using SymmetricEigenAnalysisType = itk::SymmetricEigenAnalysisFixedDimension<4,
+                                                                                 itk::Matrix<ParametersValueType, 4, 4>,
+                                                                                 vnl_vector<ParametersValueType>,
+                                                                                 vnl_matrix<ParametersValueType>>;
+    SymmetricEigenAnalysisType symmetricEigenSystem;
+
+    symmetricEigenSystem.ComputeEigenValuesAndVectors(N, eigenValues, eigenVectors);
+
+    itkDebugMacro(<< "EigenVectors " << eigenVectors);
+    itkDebugMacro(<< "EigenValues " << eigenValues);
+
+    /* By default eigen values are sorted in ascending order therefore the
+    maximum eigen value is the one in the fourth place = index 3. We need the
+    eigen vector associated with the maximum eigenvalue, so we take the
+    eigenvector from the last row, index=3 */
+
+    versor.Set(eigenVectors[3][1], eigenVectors[3][2], eigenVectors[3][3], eigenVectors[3][0]);
+    itkDebugMacro(<< "Resulting versor" << versor);
+  }
+  else
+  {
+    // Remember..
```

Comment:
Not sure this comment is necessary, or whether the else block as a whole is necessary, as it is empty. There is already a mention to the need of having at least 3 landmarks at the beginning of the `if` block.